code/game/gamemodes/malfunction/Malf_Modules.dm

```diff
@@ -172,325 +421,364 @@
 	if(C)
 		C.post_status("alert", "lockdown")
 
-	verbs -= /mob/living/silicon/ai/proc/lockdown
-	minor_announce("Hostile runtime detected in door controllers. Isolation Lockdown protocols are now in effect. Please remain calm.","Network Alert:", 1)
-	to_chat(src, "<span class = 'warning'>Lockdown Initiated. Network reset in 90 seconds.</span>")
+	minor_announce("Hostile runtime detected in door controllers. Isolation lockdown protocols are now in effect. Please remain calm.","Network Alert:", TRUE)
+	to_chat(owner, "<span class='danger'>Lockdown initiated. Network reset in 90 seconds.</span>")
 	addtimer(CALLBACK(GLOBAL_PROC, .proc/minor_announce,
 		"Automatic system reboot complete. Have a secure day.",
 		"Network reset:"), 900)
 
+
+//Destroy RCDs: Detonates all non-cyborg RCDs on the station.
 /datum/AI_Module/large/destroy_rcd
 	module_name = "Destroy RCDs"
 	mod_pick_name = "rcd"
-	description = "Send a specialised pulse to detonate all hand-held and exosuit Rapid Cconstruction Devices on the station."
+	description = "Send a specialised pulse to detonate all hand-held and exosuit Rapid Construction Devices on the station."
 	cost = 25
-	one_time = 1
-
-	power_type = /mob/living/silicon/ai/proc/disable_rcd
-
-/mob/living/silicon/ai/proc/disable_rcd()
-	set category = "Malfunction"
-	set name = "Destroy RCDs"
-	set desc = "Detonate all RCDs on the station, while sparing onboard cyborg RCDs."
-	set waitfor = FALSE
-
-	if(!canUseTopic() || malf_cooldown > world.time)
-		return
+	one_purchase = TRUE
+	power_type = /datum/action/innate/ai/destroy_rcds
+	unlock_text = "<span class='notice'>After some improvisation, you rig your onboard radio to be able to send a signal to detonate all RCDs.</span>"
+	unlock_sound = 'sound/items/timer.ogg'
+
+/datum/action/innate/ai/destroy_rcds
+	name = "Destroy RCDs"
+	desc = "Detonate all non-cyborg RCDs on the station."
+	button_icon_state = "detonate_rcds"
+	uses = 1
```

Comment:
@Xhuis This is supposed to be able to work multiple times, not just once: https://github.com/tgstation/tgstation/pull/11250
Please add more uses. The intend behind this is to prevent easy RCD replacement via autolathes and Engineering vendors.